packages-exp/auth-exp/src/core/user/user_impl.ts

```diff
@@ -100,16 +101,27 @@ export class UserImpl implements User {
     return reload(this);
   }
 
-  _updateTokensIfNecessary(response: IdTokenResponse): boolean {
+  async _updateTokensIfNecessary(
+    response: IdTokenResponse,
+    reload = false
+  ): Promise<void> {
```

Comment:
we don't need the return value anymore?

Reply:
No I refactored all of this so that it handles the persistence + notification if necessary